test_scripts/iAP2_Emulation/iAP2_Bluetooth_to_USB_transport_switch_happy_path.lua

```diff
@@ -0,0 +1,158 @@
+config.defaultProtocolVersion = 2
+
+-- [[ Required Shared Libraries ]]
+local commonFunctions = require("user_modules/shared_testcases/commonFunctions")
+local commonSteps = require("user_modules/shared_testcases/commonSteps")
+local commonTestCases = require("user_modules/shared_testcases/commonTestCases")
+
+local mobile = require("mobile_connection")
+local mobile_session = require("mobile_session")
+
+local tcp = require("tcp_connection")
+local file_connection = require("file_connection")
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_iap2_emulation")
+require("cardinalities")
+require("user_modules/AppTypes")
+
+-- [[Local variables]]
+local app_RAI_params = config["application1"].registerAppInterfaceParams
+
+--[[ 
+IN ORDER TO USE IAP2 EMULATION IN SDL IT MUST BE BUILT ON 
+https://github.com/dev-gh/sdl_core/tree/experimental/IAP_adapters_emulation
+WITH 
+BUILD_TESTS = ON
+]]
+local iAP2_BT_DeviceID = "127.0.0.1"
+local iAP2_BT_Port = 23456
+local iAP2_BT_out = "iap2bt.out"
+local iAP2_BT_Type = "BLUETOOTH"
+
+-- Device IDs must be the same in order to trigger switching logic
+local iAP2_USB_DeviceID = iAP2_BT_DeviceID
+local iAP2_USB_Port = 34567
+local iAP2_USB_out = "iap2usb.out"
+local iAP2_USB_Type = "USB_IOS"
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+local function createIAP2Device(deviceID, devicePort, deviceOut)
+    local iap2Connection = tcp.Connection(deviceID, devicePort)
+    local fileConnection = file_connection.FileConnection(deviceOut, iap2Connection)
+    local iap2Device = mobile.MobileConnection(fileConnection)
+
+    event_dispatcher:AddConnection(iap2Device)
+
+    return iap2Device
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local iap2bt_device = 0
+
+Test["Connecting iAP2 Bluetooth"] =
+    function(self)
+    -- iAP2 Bluetooth connection with application registered
+    iap2bt_device = createIAP2Device(iAP2_BT_DeviceID, iAP2_BT_Port, iAP2_BT_out)
+
+    EXPECT_HMICALL(
+        "BasicCommunication.UpdateDeviceList",
+        {
+            deviceList = {{id = config.deviceMAC, name = iAP2_BT_DeviceID, transportType = iAP2_BT_Type}}
+        }
+    ):Do(
+        function(_, data)
+            self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+        end
+    )
+
+    Test:connectMobile(iap2bt_device)
+
+    local iap2bt_mobileSession = mobile_session.MobileSession(self, iap2bt_device, app_RAI_params)
+
+    Test:startSession(iap2bt_mobileSession)
+end
+
+Test["Connecting iAP2 USB"] =
+    function(self)
+    -- iAP2 USB connection with same application
+    local iap2usb_device = createIAP2Device(iAP2_USB_DeviceID, iAP2_USB_Port, iAP2_USB_out)
+
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered"):Times(0)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered"):Times(0)
```

Comment:
@dev-gh In case of ```:Times(0)``` DelayedExp has to be used
But the one in commonTestCases module doesn't fit since it uses mobile connection for fake event.
And we changing mobile connections during the test.
So existing function has to be redesigned to use HMI connection or new function has to be created.